contributors/design-proposals/node-allocatable.md

```diff
@@ -89,61 +73,253 @@ The flag will be specified as a serialized `ResourceList`, with resources define
+Kubelet evicts pods based on QoS and user configured eviction thresholds.
+More deails in [this doc](./kubelet-eviction.md#enforce-node-allocatable)
+
+From v1.6, if `Allocatable` is enforced by default across all pods on a node using cgroups, pods cannot to exceed `Allocatable`.
+Memory and CPU limits are enforced using cgroups, but there exists no easy means to enforce storage limits though. 
+Enforcing storage limits using Linux Quota is not possible since it's not hierarchical. 
+Once storage is supported as a resource for `Allocatable`, Kubelet has to perform evictions based on `Allocatable` in addition to `Capacity`.
+
+Note that eviction limits are enforced on pods only and system daemons are free to use any amount of resources unless their reservations are enforced.
+
+Here is an example to illustrate Node Allocatable for memory:
+
+Node Capacity is `32Gi`, kube-reserved is `2Gi`, system-reserved is `1Gi`, eviction-hard is set to `<100Mi`
+
+For this node, the effective Node Allocatable is `28.9Gi` only; i.e. if kube and system components use up all their reservation, the memory available for pods is only `28.9Gi` and kubelet will evict pods once overall usage of pods crosses that threshold.
+
+If we enforce Node Allocatable (`28.9Gi`) via top level cgroups, then pods can never exceed `28.9Gi` in which case evictions will not be performed unless kernel memory consumption is above `100Mi`.
+
+In order to support evictions and avoid memcg OOM kills for pods, we will set the top level cgroup limits for pods to be `Node Allocatable` + `Eviction Hard Tresholds`.
+
+However, the scheduler is not expected to use more than `28.9Gi` and so `Node Allocatable` on Node Status will be `28.9Gi`.
+
+If kube and system components do not use up all their reservation, with the above example, pods will face memcg OOM kills from the node allocatable cgroup before kubelet evictions kick in.
+To better enforce QoS under this situation, Kubelet will apply the hard eviction thresholds on the node allocatable cgroup as well, if node allocatable is enforced.
+The resulting behavior will be the same for user pods. 
+With the above example, Kubelet will evict pods whenever pods consume more than `28.9Gi` which will be `<100Mi` from `29Gi` which will be the memory limits on the Node Allocatable cgroup.
+
+## General guidelines
+
+System daemons are expected to be treated similar to `Guaranteed` pods.
+System daemons can burst within their bounding cgroups and this behavior needs to be managed as part of kubernetes deployment.
+For example, Kubelet can have its own cgroup and share `KubeReserved` resources with the Container Runtime.
+However, Kubelet cannot burst and use up all available Node resources if `KubeReserved` is enforced.
+
+Users are adviced to be extra careful while enforcing `SystemReserved` reservation since it can lead to critical services being CPU starved or OOM killed on the nodes.
```

Comment:
Since so many OS flavors run with systemd-logind it's also worth calling out that implicitly any resource consumption associated with a user session (ssh into a node for example) is unbounded.  Operators should account for that in there configured value for system-reserved even if not enforced.